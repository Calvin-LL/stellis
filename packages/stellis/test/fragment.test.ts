import { describe, it, expect } from "vitest";
import transformStellis from "./utils";

describe('fragment', () => {
  it('should transform fragment with static child element', async () => {
    const code = `
      <>
        <h1>Hello World</h1>
        <h1>Hello World</h1>
      </>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform fragment with dynamic child element', async () => {
    const code = `
      <>
        {dynamicChild()}
        {dynamicChild()}
      </>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});