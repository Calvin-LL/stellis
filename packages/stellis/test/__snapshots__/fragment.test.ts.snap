// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fragment > should transform fragment with dynamic child element 1`] = `"() => [dynamicChild(), dynamicChild()];"`;

exports[`fragment > should transform fragment with static child element 1`] = `
"import { $$html as _$$html } from \\"stellis\\";
const _template = \\"<h1>Hello World</h1>\\",
  _template2 = \\"<h1>Hello World</h1>\\";
[_$$html(_template, true), _$$html(_template2, true)];"
`;
